sql/parser/sql.y

```diff
@@ -1190,21 +1199,38 @@ table_constraint:
 
 constraint_elem:
   CHECK '(' a_expr ')' {}
-| UNIQUE '(' name_list ')'
+| UNIQUE '(' name_list ')' opt_storing
   {
-    $$ = &IndexTableDef{Unique: true, Columns: NameList($3)}
+    $$ = &IndexTableDef{Unique: true, Columns: NameList($3), Storing: $5}
   }
-| UNIQUE existing_index {}
 | PRIMARY KEY '(' name_list ')'
   {
     $$ = &IndexTableDef{PrimaryKey: true, Unique: true, Columns: NameList($4)}
   }
-| PRIMARY KEY existing_index {}
 | EXCLUDE '(' exclusion_constraint_list ')'
     exclusion_where_clause {}
 | FOREIGN KEY '(' name_list ')' REFERENCES qualified_name
     opt_column_list key_match key_actions {}
 
+storing:
+  COVERING
+| STORING
+
+// You might be tempted to add syntax like STORING ALL or STORING (*)
+// to indicate that an index show store all of the columns in a
+// table. Doing so is problematic because such a specification would
+// make adding a column which more expensive as we would have to
+// rewrite every index that contained a STORING (*) specification.
```

Comment:
I thought about this limitation while sitting through curriculum night at the girls' school. I think I can change the way we encode values in indexes such that they are a list of `<columnID, value>` pairs which would allow a NULL-able column to be added without touching the index values. Similarly dropping a column would just drop it from the metadata as long as the column was only be stored in the index and not used for the index key itself.

But this is a bit more substantial than I'd like to tackle in this PR. I'm going to change the above comment into a TODO.